packages/repository/test/unit/repositories/constraint-utils.unit.ts

```diff
@@ -0,0 +1,113 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {
+  FilterBuilder,
+  Filter,
+  Fields,
+  Where,
+  WhereBuilder,
+} from '../../../src/query';
+import {AnyObject} from 'loopback-datasource-juggler';
+import {
+  constrainFilter,
+  constrainWhere,
+} from '../../../src/repositories/constraint-utils';
+
+describe('constraint utility functions', () => {
+  let originalFilter: Filter = {};
+  let originalWhere: Where = {};
+  const constraint: AnyObject = givenAWhereConstraint();
+
+  before(() => {
+    originalFilter = givenAFilter({a: true}, {x: 'x'}, 5);
+    originalWhere = givenAWhere();
+  });
+  context('constrainFilter', () => {
+    it('applies a where constraint', () => {
+      const result = constrainFilter(originalFilter, constraint);
+      expect(result).to.deepEqual({
+        fields: {a: true},
```

Comment:
When reading this test in isolation, it's not clear where `{a:true}` is coming from.

If the test depends on specific values passed to `constrainFilter`, then these specific values need to be provided in the test.

Alternatively, refer to `originalFilter` properties in the assertion, e.g.

```
expect(result).to.deepEqual({
  fields: originalFilter.fields,
  limit: originalFilter.limit,
  where: Object.assign({}, originalFilter.where, {id: '5'}),
});
```

Reply:
Agreed :) done.